R/helper_scrape_gc.R

```diff
@@ -8,302 +8,240 @@
-
-      if (gameId %in% c("2000_03_SD_KC", "2000_06_BUF_MIA", "1999_01_BAL_STL")) {
-        warning(warn <- 1)
-      }
-
-      season <- as.integer(substr(gameId, 1, 4))
-
-      if (is.null(dir)) {
-        path <- "https://raw.githubusercontent.com/nflverse/nflfastR-raw/master/raw"
-
-        if(isFALSE(qs)) fetched <- curl::curl_fetch_memory(glue::glue("{path}/{season}/{gameId}.rds"))
-
-        if(isTRUE(qs)) fetched <- curl::curl_fetch_memory(glue::glue("{path}/{season}/{gameId}.qs"))
+  season <- as.integer(substr(gameId, 1, 4))
 
-        if (fetched$status_code == 404) {
-          warning(warn <- 3)
-        } else if (fetched$status_code == 500) {
-          warning(warn <- 4)
-        }
+  raw <- fetch_raw(game_id = gameId, dir = dir)
 
-        if(isFALSE(qs)) raw <- read_raw_rds(fetched$content)
+  game_json <- raw[[1]]
 
-        if(isTRUE(qs)) raw <- qs::qdeserialize(fetched$content)
-
-      } else {
-        # build path to locally stored game files. This functionality is primarily
-        # for the data repo maintainer
-        if(isFALSE(qs)) p <- glue::glue("{dir}/{season}/{gameId}.rds")
-        if(isTRUE(qs)) p <- glue::glue("{dir}/{season}/{gameId}.qs")
-
-        if (file.exists(p) == FALSE) {
-          warning(warn <- 5)
-        }
-
-        if(isFALSE(qs)) raw <- readRDS(p)
-        if(isTRUE(qs)) raw <- qs::qread(p)
-      }
-
-      game_json <- raw[[1]]
-
-      date_parse <- names(raw)[1] %>% stringr::str_extract(pattern = "[0-9]{8}")
-      date_year <- stringr::str_sub(date_parse, 1, 4)
-      date_month <- stringr::str_sub(date_parse, 5, 6)
-      date_day <- stringr::str_sub(
-        date_parse, nchar(date_parse) - 1,
-        nchar(date_parse)
-      )
+  date_parse <- names(raw)[1] %>% stringr::str_extract(pattern = "[0-9]{8}")
+  date_year <- stringr::str_sub(date_parse, 1, 4)
+  date_month <- stringr::str_sub(date_parse, 5, 6)
+  date_day <- stringr::str_sub(
+    date_parse, nchar(date_parse) - 1,
+    nchar(date_parse)
+  )
 
-      week <- as.integer(substr(gameId, 6, 7))
-      if (week <= 17) {
-        season_type <- "REG"
-      } else {
-        season_type <- "POST"
-      }
+  week <- as.integer(substr(gameId, 6, 7))
+  if (week <= 17) {
+    season_type <- "REG"
+  } else {
+    season_type <- "POST"
+  }
 
-      if (date_year < 1999) {
-        warning(warn <- 2)
-      }
+  if (date_year < 1999) {
+    cli::cli_abort("You asked a game from {date_year}, but data only goes back to 1999.")
+  }
 
-      # excluding last element since it's "crntdrv" and not an actual
-      drives <- game_json$drives[-length(game_json$drives)]
+  # excluding last element since it's "crntdrv" and not an actual
+  drives <- game_json$drives[-length(game_json$drives)]
+
+  # list of plays
+  # each play has "players" column which is a list of player stats from the play
+  plays <- suppressWarnings(furrr::future_map_dfr(seq_along(drives), function(x) {
+    cbind(
+      "drive" = x,
+      data.frame(do.call(
```

Comment:
hmm, it’s usually good practice to swap do.call rbind out to datatable rbindlist or dplyr bind_rows esp since both options are available 

https://rstudio-pubs-static.s3.amazonaws.com/406521_7fc7b6c1dc374e9b8860e15a699d8bb0.html

Reply:
I plan to adjust some old code in the parser later. It's beyond the scope of this PR. All I changed in that file is removing tryCatch

Reply:
Ahh I misread the scope